File: 961_Nrepeats_2N_array.py
# -*- coding: utf-8 -*-
"""
Created on Mon Sep 23 09:49:51 2019

@author: DinoBob
"""

class Solution:
    def repeatedNTimes(self, A):
        i = 0
        while i + 3 < len(A):
            ans = self.find_repeat(A[i: i+4])
            if ans!= None:
                return ans
            i += 4
        if i < len(A):
            return self.find_repeat(A[i:])
            
            
    def find_repeat(self, subA):
        arr = []
        for i in subA:
            if i in arr:
                return i
            else:
                arr.append(i)
        
ans = Solution()
ans.repeatedNTimes([4,1,7,0,0,9,0,0])

File: Leetcode/src/961_Nrepeats_2N_array.java
class Solution {
    public Integer repeatedNTimes(int[] A) {
        int k = 0;
        while((k + 3) < A.length){
            Integer ans = findRepeat(A, k, k+4);
            if(ans != null){
                return ans;
            }
            k +=4;
        }
        if (k < A.length){
            return findRepeat(A, k, A.length);
        } 
        return null;
    }
    
        
    public Integer findRepeat(int[] arr, int begin, int end){
        for(int i = begin; i < end - 1; i++){
            for(int j = i+1; j < end; j++){
                if(arr[i] == arr[j]){
                    return arr[i];
                }
            }
        }
        return null;        
    }
}